lib/istio/addon/rule-detail/route.js

```diff
@@ -0,0 +1,23 @@
+import Route from '@ember/routing/route';
+import { get } from '@ember/object';
+import { hash } from 'rsvp';
+import { inject as service } from '@ember/service';
+
+export default Route.extend({
+  catalog: service(),
+
+  beforeModel() {
+    return get(this, 'catalog').fetchUnScopedCatalogs();
+  },
+
+  model(params) {
+    const store = get(this, 'store');
+
+    return hash({ rule: store.find('app', get(params, 'rule_id')), });
```

Comment:
If we don't need a hash promise here prefer `store.find` with a `then` that returns the response in an object.